File: iOS5bookExamples/ch36p901peopleGroups/p740peopleGroups/Person.m


#import "Person.h"


@implementation Person
@synthesize firstName, lastName;


- (void)encodeWithCoder:(NSCoder *)encoder {
    //[super encodeWithCoder: encoder]; // not in this case
    [encoder encodeObject:self->lastName forKey:@"last"];
    [encoder encodeObject:self->firstName forKey:@"first"];
}

- (id) initWithCoder:(NSCoder *)decoder {
    //self = [super initWithCoder: decoder]; // not in this case
    self = [super init];
    self->lastName = [decoder decodeObjectForKey:@"last"];
    self->firstName = [decoder decodeObjectForKey:@"first"];
    return self;
}


@end

File: iOS9bookExamples/bk2ch23p809peopleGroups/ch36p1066peopleGroups/Person.swift


import UIKit

class Person: NSObject, NSCoding {
    var firstName : String
    var lastName : String
    
    override var description : String {
        return self.firstName + " " + self.lastName
    }
    
    init(firstName:String, lastName:String) {
        self.firstName = firstName
        self.lastName = lastName
        super.init()
    }
    
    func encodeWithCoder(coder: NSCoder) {
        // do not call super in this case
        coder.encodeObject(self.lastName, forKey: "last")
        coder.encodeObject(self.firstName, forKey: "first")
    }
    
    required init(coder: NSCoder) {
        self.lastName = coder.decodeObjectForKey("last") as! String
        self.firstName = coder.decodeObjectForKey("first") as! String
        // do not call super init(coder:) in this case
        super.init()
    }

}
